### PR TITLE
fix: CFAST snippets generation

### PR DESCRIPTION
**Pull request by ishche**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java [hunk 1]
```diff
@@ -243,15 +253,24 @@ private Location convertLocation(Node node) {
...
+      return "<snippet creation error>";
+    }
+    List<CobolDocumentModel.Line> lines = doc.getLines();
     StringBuilder sb = new StringBuilder();
-    for (int i = 0; i < SNIPPET_LENGTH && i < lines.length; i++) {
-      if (i > 0) {
+    int startLine = node.getLocality().getRange().getStart().getLine();
+    int stopLine = node.getLocality().getRange().getEnd().getLine() - startLine + 1 > SNIPPET_LENGTH
+            ? startLine + SNIPPET_LENGTH
+            : node.getLocality().getRange().getEnd().getLine() + 1;
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java]: Math.min

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/extendedapi/Paragraph.java [hunk 2]
```diff
@@ -24,7 +24,7 @@ public class Paragraph extends CFASTNode {
   String snippet;
   String name;
 
-  public Paragraph(String snippet, String name, Location location) {
+  public Paragraph(String name, String snippet, Location location) {
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/extendedapi/Paragraph.java:27]: not sure what the point of this change was...

Comment (ishche) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/extendedapi/Paragraph.java:27]: So it aligns with the Section constructor's argument order. To reduce cognitive load :)

Comment (slavek-kucera) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/extendedapi/Paragraph.java:27]: Are you sure you changed all of them (the function signature is the same)?

Comment (ishche) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/extendedapi/Paragraph.java:27]: I think so, I would not expect other snippets in CCF.

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**